lib/service/feralfile_service.dart

```diff
@@ -234,41 +241,101 @@ class FeralFileServiceImpl extends FeralFileService {
     return exhibitionResponse.result;
   }
 
-  Future<List<Artwork>> _getExhibitionArtworkByDirectApi(String exhibitionId,
-      {bool withSeries = false}) async {
-    final artworks =
-        await _feralFileApi.getListArtworks(exhibitionId: exhibitionId);
-    List<Artwork> listArtwork = artworks.result;
+  Future<FeralFileListResponse<Artwork>> _getExhibitionArtworkByDirectApi(
+      String exhibitionId,
+      {bool withSeries = false,
+      int? offset,
+      int? limit}) async {
+    FeralFileListResponse<Artwork> artworksResponse =
+        await _feralFileApi.getListArtworks(
+            exhibitionId: exhibitionId, offset: offset, limit: limit);
     log
       ..info(
         '[FeralFileService] [_getExhibitionByDirectApi] '
-        'Get exhibition artworks: ${listArtwork.length}',
+        'Get exhibition artworks: ${artworksResponse.result.length}',
       )
       ..info(
         '[FeralFileService] [_getExhibitionByDirectApi] '
         'Get exhibition artworks: '
-        '${listArtwork.map((e) => e.id).toList()}',
+        '${artworksResponse.result.map((e) => e.id).toList()}',
       );
     if (withSeries) {
       final listSeries = await getListSeries(exhibitionId);
       final seriesMap =
           Map.fromEntries(listSeries.map((e) => MapEntry(e.id, e)));
-      listArtwork = listArtwork
-          .map((e) => e.copyWith(series: seriesMap[e.seriesID]))
-          .toList();
+      artworksResponse = artworksResponse.copyWith(
+          result: artworksResponse.result
+              .map((e) => e.copyWith(series: seriesMap[e.seriesID]))
+              .toList());
     }
-    return listArtwork;
+    return artworksResponse;
   }
 
-  Future<List<Artwork>> _getExhibitionFakeArtworks(String exhibitionId) async {
+  Future<FeralFileListResponse<Artwork>> _getExhibitionFakeArtworks(
+      String exhibitionId,
+      {required int offset,
+      required int limit}) async {
+    final cacheArtworks = await _getCacheFakeArtworks(exhibitionId);
+    final total = cacheArtworks.length;
+    final List<Artwork> listArtworks =
+        cacheArtworks.sublist(offset, min(offset + limit, total));
+    return FeralFileListResponse(
+        result: listArtworks,
+        paging: Paging(offset: offset, limit: limit, total: total));
+  }
+
+  Future<List<Artwork>> _getCacheFakeArtworks(String exhibitionId) async {
+    if (_cachedFakeArtwork.containsKey(exhibitionId)) {
```

Comment:
Why do we need to cache fake artworks?

Reply:
To simply our logic, we get fake exhibition artwork by query all series then create fake artworks of each serires, caching will save us alot of unecessary fetching